core/block_context.go

```diff
@@ -0,0 +1,87 @@
+package core
+
+import (
+	"math/big"
+
+	"github.com/celo-org/celo-blockchain/common"
+	"github.com/celo-org/celo-blockchain/contract_comm/blockchain_parameters"
+	"github.com/celo-org/celo-blockchain/contract_comm/currency"
+	gpm "github.com/celo-org/celo-blockchain/contract_comm/gasprice_minimum"
+	"github.com/celo-org/celo-blockchain/core/types"
+	"github.com/celo-org/celo-blockchain/core/vm"
+)
+
+// BlockContext represents contextual information about the blockchain state
+// for a give block
+type BlockContext interface {
+	// GetGoldGasPriceMinimum retrieves the gas price minimum for the CELO token
+	GetGoldGasPriceMinimum() *big.Int
+
+	// GetGasPriceMinimum retrieves the gas price minimum for any currency
+	// Also indicates if the currency is not whitelisted
+	GetGasPriceMinimum(feeCurrency *common.Address) (gpm *big.Int, isWhitelisted bool)
```

Comment:
I like having an explicit error condition here, but I lean towards having a separate `IsWhiteListedFeeCurrency` function

Reply:
I did have that function first... but the usage pattern the get more cumbersome...

Let's say you have `IsWhiteListedFeeCurrency`, what signature will you use for `GetGasPriceMinimun()`?

If you you call `GetGasPriceMinimun()` on a currency that is not whitelisted. What do you do?

1. panic
2. return defaultValue
3. change signature so that `GetGasPriceMinimun()` returns also an error, and define the `NotWhiteListedError`

Neither of those sounds good to me. (1) and (2) are defintely out of the question... and 3 implies that now every caller needs to check an error that can only happen if the caller didn't first check `IsWhitelisted...`

So, i changed it to be a single function.

Reply:
I actually agree with this and was thinking approach 3 would be best. I reference this in another comment where the lack of an `IsWhiteListed` results in a behavior change. My main reason for wanting to reintroduce the `IsWhiteListed` is that with this interface it is currently impossible to tell when a currency _is_ whitelisted, but the GPM was not available for an unrelated reason. (e.g. no oracle reports)

Reply:
I was thinking of #3 (switch the bool to error)

Reply:
removed gpm check; kept only isWhitelisted